recipes/cad1/task1/baseline/enhance.py

```diff
@@ -395,15 +469,11 @@ def enhance(config: DictConfig) -> None:
             / f"{listener_info['name']}_{song_name}_remix.wav"
         )
 
-        # Clip and save
-        if config.soft_clip:
-            enhanced = np.tanh(enhanced)
-        n_clipped = np.sum(np.abs(enhanced) > 1.0)
+        # clip and save enhanced signal
+        clipped_signal, n_clipped = clip_signal(enhanced, config.soft_clip)
         if n_clipped > 0:
             logger.warning(f"Writing {filename}: {n_clipped} samples clipped")
-        np.clip(enhanced, -1.0, 1.0, out=enhanced)
```

Comment:
This is duplicated code. We have a write_signal that handles clipping and conversion to 16bit.  The only bit you might need to do would be the

Something like

```
if config.soft_clip:
  signal = tanh(signal)
write_signal(signal, filename)
```

You can then remove the clip_signal, to_16bit and resample functions to avoid duplicating code.

The read_signal write_signal  methods were recently refactored in the main branch. So if the plan is to release this as 0.3.2 it would be worth copying them over from main now.

I think it's important that we try to reuse our library code in the recipes wherever possible so that others can see by example.

Reply:
Agree with reusing code. I think it should be possible to `git cherry-pick` the merge commit from `main` that incorporated these changes on that branch to pull them all over.

Reply:
Ok, I didn't reuse these functions as they are quite "new" in main.
I wasn't sure how to deal with all these defferences between 0.3 and main

Reply:
Ok, I didn't reuse these functions as they are quite "new" in main.
I wasn't sure how to deal with all these defferences between 0.3 and main

Reply:
@ns-rse, I always confused about the cherry-pick.
What would be better in this case, `charry-pick` or just add another new commit copying these function?

Reply:
@ns-rse, I always confused about the cherry-pick.
What would be better in this case, `cherry-pick` or just add another new commit copying these function?

Reply:
@ns-rse, I always confused about the cherry-pick.
What would be better in this case, `charry-pick` or just add another new commit copying these function?

Reply:
I don't have a particularly strong opinion, I think it depends whether you want to have _everything_ tracked in the `git` history, which `cherry-pick` should preserve.

If its straight-forward to copy code manually from another branch and commit and you're comfortable doing that it would also be fine.